frontend/src/views/producer/producer-modal/index.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import Modal, { Content } from '../../../components/modal';
+import Typography from '../../../components/typography';
+import { connect } from '../../../utils';
+import { withTheme } from 'styled-components';
+import Icon from 'antd/lib/icon';
+
+export default withTheme(
+  connect('store')(
+    class AppModal extends Component {
+      state = {
+        show: false,
+        handledSignalToShow: false,
+      };
+      constructor(props) {
+        super(props);
+        if (props.store.ui.auth.validateTokenFailed) {
+          this.state = { show: true };
+          window.setTimeout(() => {
+            try {
+              props.store.ui.auth.clearValidationError();
+            } catch (error) {
+              console.error(error);
+            }
+          }, 5000);
+        }
+      }
+      componentWillUpdate() {
+        // only show modal if its 0.5s pass waiting.
+        // this is to avoid flashing effects
+        if (
+          this.props.store.ui.auth.validateTokenInProgress &&
+          !this.state.handledSignalToShow
+        ) {
+          this.setState(
+            {
+              handledSignalToShow: true,
+            },
+            () => {
+              new Promise((resolve, reject) => {
+                window.setTimeout(resolve, 300);
```

Comment:
These multiple timeouts setting different states makes it really hard to read what happens here.. feels a bit hackish implementation to me..

Reply:
well i suppose i can reduce it to use setTimeout only instead of a Promise